Update documentation on new way of visualising layers on Kedro-viz (#2771)
kedro-org/kedro: docs/source/visualisation/kedro-viz_visualisation.md

For example, the [data engineering convention](https://towardsdatascience.com/the-importance-of-layered-thinking-in-data-engineering-a09f685edc71) labels datasets according to the stage of the pipeline (e.g. whether the data has been cleaned).

> As of Kedro version 18.9, we've made a change to the way layers are defined in the Data Catalog. Instead of specifying the layer attribute directly within a dataset's definition, it is now included under the kedro-viz metadata. 

Reviewer: ```suggestion
As of Kedro version 18.9, we've made a change to the way layers are defined in the Data Catalog. The definition is now included under the kedro-viz metadata, where previously it was an attribute specified within a dataset's definition. 
```

> 

Reviewer: Please can you add a sentence to introduce the code to indicate this is the new specification. Something like

```suggestion
Here is how to include the layer definition using Kedro-Viz metadata:
```

```yaml
companies:
  type: pandas.CSVDataSet
  filepath: data/01_raw/companies.csv
  metadata: 
    kedro-viz:
      layer: raw
```

> Before Kedro 18.9, layers were specified directly within a dataset's definition in the Data Catalog. Here's an example of the old method:

Reviewer: ```suggestion
Prior to Kedro 18.9, layers were specified directly within a dataset's definition in the Data Catalog:
```

```yaml
companies:
  type: pandas.CSVDataSet
  filepath: data/01_raw/companies.csv
  layer: raw
```

> However, it is essential to be aware that this old way of defining layers will no longer be supported from Kedro version 19.0. Therefore, we highly recommend adapting to the new method where layers are specified under the kedro-viz metadata in the Data Catalog

Reviewer: ```suggestion
The previous way of defining layers will **no longer be supported** from Kedro version 19.0. From that version onwards, you need to modify your `catalog.yml` to use the new method and specify layers as metadata.
```

> Open `catalog.yml` for the completed spaceflights tutorial and replace the existing code with the following:

Reviewer: I think it would be better to specify this as a diff rather than new code. Will we need to update the starter too?

Author: i didn't find any layer reference in the starter docs. I did find in the glossary which I also updated.

Reviewer: Oh, sorry, I meant we will have to update the starter code if it generates it with layers in place but I have checked now and realise that it doesn't, so no update needed. Thanks for changing the glossary.

For example, the [data engineering convention](https://towardsdatascience.com/the-importance-of-layered-thinking-in-data-engineering-a09f685edc71) labels datasets according to the stage of the pipeline (e.g. whether the data has been cleaned).

> As of Kedro version 18.9, we've made a change to the way layers are defined in the Data Catalog. The definition is now included under the kedro-viz metadata, where previously it was an attribute specified within a dataset's definition. 

Reviewer 2: I'm wondering if we could rephrase this part as just pretending this has been like that forever, and then adding a section or callout about how to migrate old catalogs. I think the version of Kedro that introduced the feature is not very interesting for many readers (and it becomes less and less interesting the more time it passes, hence it becomes documentation debt). What are your thoughts @stichbury?

Reviewer: I'm not sure about gaslighting our users 😆 

I have made a commit to change the way we describe the change based on a conversation on Slack.

For example, the [data engineering convention](https://towardsdatascience.com/the-importance-of-layered-thinking-in-data-engineering-a09f685edc71) labels datasets according to the stage of the pipeline (e.g. whether the data has been cleaned).

In Kedro version 0.18.9 we changed the way layers are defined in the Data Catalog. The definition is now included under the `metadata` key for `kedro-viz` (previously it was an attribute specified within a dataset's definition).

> Here's an example of how to use the kedro-viz metadata to define layers:

Reviewer 3: ```suggestion
Here's an example of how to use the Kedro-Viz metadata to define layers:
```

```yaml
companies:
  type: pandas.CSVDataSet
  filepath: data/01_raw/companies.csv
  metadata:
    kedro-viz:
      layer: raw
```

In earlier versions of Kedro, layers were specified within a dataset's definition in the Data Catalog, but this will **no longer be supported** from Kedro version 0.19.0. From that version onwards, your `catalog.yml` must specify layers as metadata.

```diff
companies:
  type: pandas.CSVDataSet
  filepath: data/01_raw/companies.csv
-  layer: raw
+   metadata:
+     kedro-viz:
+       layer: raw
```

Open `catalog.yml` for the completed spaceflights tutorial and define layers in the following way:

```yaml
companies:
  type: pandas.CSVDataSet
  filepath: data/01_raw/companies.csv
  metadata:
    kedro-viz:
      layer: raw

reviews:
  type: pandas.CSVDataSet
  filepath: data/01_raw/reviews.csv
  metadata:
    kedro-viz:
      layer: raw

shuttles:
  type: pandas.ExcelDataSet
  filepath: data/01_raw/shuttles.xlsx
  metadata:
    kedro-viz:
      layer: raw

preprocessed_companies:
  type: pandas.ParquetDataSet
  filepath: data/02_intermediate/preprocessed_companies.pq
  metadata:
    kedro-viz:

>       layer: raw

Reviewer 4: Shouldn't this layer also still be intermediate?

preprocessed_shuttles:
  type: pandas.ParquetDataSet
  filepath: data/02_intermediate/preprocessed_shuttles.pq
  metadata:
    kedro-viz:

>       layer: raw

Reviewer 4: Shouldn't this layer also still be intermediate?

model_input_table:
  type: pandas.ParquetDataSet
  filepath: data/03_primary/model_input_table.pq
  metadata:
    kedro-viz:

>       layer: raw

Reviewer 4: Shouldn't this layer also still be primary?

regressor:
  type: pickle.PickleDataSet
  filepath: data/06_models/regressor.pickle
  versioned: true
  metadata:
    kedro-viz:

>       layer: raw

Reviewer 4: Shouldn't this layer also still be models?
